### PR TITLE
Fix/slim down core dependencies

### PR DESCRIPTION
**Pull request by damianlegawiec**
- [X] require only Rails gems that we need
- [X] require sprockets only for backend/frontend
- [X] require responders only for backend/frontend/api
- [X] remove premailer gem from dependencies

### CODE REVIEW

File: backend/lib/spree_backend.rb [hunk 1]
```diff
@@ -1,3 +1 @@
 require 'spree/backend'
-require 'sprockets/rails'
-require 'bootstrap'
```
Comment (MatthewKennedy) [inline backend/lib/spree_backend.rb:3]: I noticed `bootstrap` and `sprockets/rails` removed from the required list on the backend, could this be done on the front end also?

```ruby
require 'spree_core'
require 'spree_api'

require 'glyphicons'
require 'inline_svg'
require 'canonical-rails'
require 'jquery-rails'
require 'turbolinks'
require 'responders'

require 'spree/frontend/middleware/seo_assist'
require 'spree/frontend/engine'
```

Comment (MatthewKennedy) [reply backend/lib/spree_backend.rb:3]: NOTE:  poppers js is also removed from this list, all seems to work fine on my localhost:3000.

Comment (damianlegawiec) [reply backend/lib/spree_backend.rb:3]: @MatthewKennedy that bootstrap requirement statement was duplicated - https://github.com/spree/spree/pull/11041/files#diff-cb942b864401de208e3140b508d1d50741dbb927a7e28b1b31bf240beb9cf073R10

Comment (damianlegawiec) [reply backend/lib/spree_backend.rb:3]: Same as the popper_js

**Review by MatthewKennedy — APPROVED** [review]

**Review by mad-eel — APPROVED** [review]

**Pull request merged**